packages/video_player/ios/Classes/VideoPlayerPlugin.m

```diff
@@ -162,6 +172,14 @@ - (void)observeValueForKeyPath:(NSString*)path
     if ([[_player currentItem] isPlaybackLikelyToKeepUp]) {
       [self updatePlayingState];
     }
+  } else if (context == playbackBufferEmptyContext) {
+    if (_eventSink != nil) {
+      _eventSink(@{@"event" : @"bufferingStart"});
+    }
+  } else if (context == playbackBufferFullContext) {
+    if (_eventSink != nil) {
+      _eventSink(@{@"event" : @"bufferingEnd"});
+    }
```

Comment:
This doesn't quite work. Running the example app on my iOS device, the circular progress indicator is activated, but then stays active throughout playback.

I guess if the video fits in the buffer, the `playbackBufferFull` property never changes. So we probably have to fire the `bufferingEnd` event also on `playbackLikelyToKeepUp`.

Reply:
Alright, gonna add that.